Documentation/api.md

```diff
@@ -0,0 +1,193 @@
+
+## PrometheusStatus
+
+Most recent observed status of the Prometheus cluster. Read-only. Not included when requesting from the apiserver, only from the Prometheus Operator API itself. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#spec-and-status
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| paused | Represents whether any actions on the underlaying managed objects are being performed. Only delete actions will be performed. | `bool` | true |
+| replicas | Total number of non-terminated pods targeted by this Prometheus deployment (their labels match the selector). | `int32` | true |
+| updatedReplicas | Total number of non-terminated pods targeted by this Prometheus deployment that have the desired version spec. | `int32` | true |
+| availableReplicas | Total number of available pods (ready for at least minReadySeconds) targeted by this Prometheus deployment. | `int32` | true |
+| unavailableReplicas | Total number of unavailable pods targeted by this Prometheus deployment. | `int32` | true |
+
+## ServiceMonitor
+
+ServiceMonitor defines monitoring for a set of services.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard object’s metadata. More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata | `metav1.ObjectMeta` | false |
+| spec | Specification of desired Service selection for target discrovery by Prometheus. | `ServiceMonitorSpec` | true |
+
+## ServiceMonitorList
+
+A list of ServiceMonitors.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| metadata | Standard list metadata More info: http://releases.k8s.io/HEAD/docs/devel/api-conventions.md#metadata | `metav1.ListMeta` | false |
+| items | List of ServiceMonitors | `[]*ServiceMonitor` | true |
+
+## ServiceMonitorSpec
+
+ServiceMonitorSpec contains specification parameters for a ServiceMonitor.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| jobLabel | The label to use to retrieve the job name from. | `string` | false |
+| endpoints | A list of endpoints allowed as part of this ServiceMonitor. | `[]Endpoint` | false |
+| selector | Selector to select Endpoints objects. | `metav1.LabelSelector` | true |
+| namespaceSelector | Selector to select which namespaces the Endpoints objects are discovered from. | `NamespaceSelector` | false |
+
+## StorageSpec
+
+StorageSpec defines the configured storage for a group Prometheus servers.
+
+| Field | Description | Scheme | Required |
+| ----- | ----------- | ------ | -------- |
+| class | Name of the StorageClass to use when requesting storage provisioning. More info: https://kubernetes.io/docs/user-guide/persistent-volumes/#storageclasses | `string` | true |
+| selector | A label query over volumes to consider for binding. | `*metav1.LabelSelector` | true |
```

Comment:
I have been thinking of this and most of the time we can actually get a reliable link, but I haven't implemented that, as I thought we could do that as a second step. Let me know if you want me to add that now.